File: src/10-19/12.py
TARGET = 500
def primes(limit=0,p=[2,3]): # p act as a static variable here.
    i = p[-1]
    while p[-1]<=limit:
        i+=2
        if all(i%j for j in range(2,i)):
            p.append(i)
    return p
i = j = 0
while 1:
    j += 1;i += j;num = i;count = 1
    for p in primes(num**0.5):
        exponent = 0
        while num % p == 0:
            exponent += 1
            num //= p
        count *= exponent + 1
    if count > TARGET:
        print(i) # 76576500
        break

File: src/10-19/12.c
#include <stdio.h>
#include <math.h>

#define TARGET 500

int main() {
    int i = 0;
    int j = 0;
    while (1) {
        j += 1;
        i += j;
        int num = i;
        int count = 1;
        for (int p = 2; p <= sqrt(i); p++) {
            int exponent = 1;
            while (num % p == 0) {
                exponent += 1;
                num /= p;
            }
            count *= exponent;
        }
        if (count > TARGET) {
            printf("%d\n", i); // 76576500
            break;
        }
    }
    return 0;
}
